src/modules/Popup/Popup.js

```diff
@@ -0,0 +1,244 @@
+    } else if (_.includes(positions, 'bottom')) {
+      style.top = Math.round(this.coords.bottom + pageYOffset)
+      style.bottom = 'auto'
+    } else {  // if not top nor bottom, we are vertically centering the element
+      const yOffset = (this.coords.height + this.popupCoords.height) / 2
+      style.top = Math.round(this.coords.bottom + pageYOffset - yOffset)
+      style.bottom = 'auto'
+
+      const xOffset = this.popupCoords.width + 8
+      if (_.includes(positions, 'right')) {
+        style.right -= xOffset
+      } else {
+        style.left -= xOffset
+      }
+    }
+
+    return style
+  }
+
+  // returns an adjusted position according to a given style
+  // so the popover always appear fully visible
+  correctPositionForViewPort(style) {
+    const { clientHeight } = document.documentElement
+    const { pageYOffset, pageXOffset } = window
+
+    if (_.isNumber(style.right)) {
+      if (style.right + pageXOffset < 0) {
+        return 'left center'
+      }
+    } else if (style.left - pageXOffset < 0) {
+      return 'right center'
+    }
+
+    if (_.isNumber(style.bottom)) {
+      if (style.bottom + this.popupCoords.height > clientHeight - pageYOffset) {
+        return 'bottom center'
+      }
+    } else if (style.top + this.popupCoords.height > pageYOffset + clientHeight) {
+      return 'top center'
+    }
+  }
+
+  getPopupStyle() {
+    if (!this.coords || !this.popupCoords) return
+    let position = this.props.position
+    let style = this.computePopupStyle(position)
+
+    // Lets detect if the popup is out of the viewport and adjust
+    // the position accordingly
+    const correctedPosition = this.correctPositionForViewPort(style)
+    if (correctedPosition) {
+      style = this.computePopupStyle(correctedPosition)
+      // Set the position so the carret will appear at the right place
+      position = correctedPosition
+    }
+
+    // Append 'px' to every numerical values in the style
+    style = _.mapValues(style, value => _.isNumber(value) ? value + 'px' : value)
+
+    this.setState({ style, position })
+  }
+
+  getPortalProps() {
+    const portalProps = {onOpen: this.onOpen}
+
+    switch(this.props.on){
+      case 'click':
+        portalProps.openOnTriggerClick = true
+        portalProps.closeOnTriggerClick = true
+        portalProps.closeOnDocumentClick = true
+        break
+      }
+      case 'focus':
+        portalProps.openOnTriggerFocus = true
+        portalProps.closeOnTriggerBlur = true
+        break
+      }
+      default:  // default to hover
+        portalProps.openOnTriggerMouseOver = true
+        portalProps.closeOnTriggerMouseLeave = true
+        // Taken from SUI: https://git.io/vPmCm
+        portalProps.mouseLeaveDelay = 70
+        portalProps.mouseOverDelay = 50
+        break
+      }
+    }
+
+    if (this.props.hoverable) {
+      portalProps.closeOnPortalMouseLeave = true
+      portalProps.mouseLeaveDelay = 300
+    }
+
+    this.setState({ portalProps })
```

Comment:
Indeed, this commit was a bit on the hasty side with no linter on... might be living a bit too dangerously right here 😄